cpp/dolfinx/io/XDMFFile.cpp

```diff
@@ -118,6 +118,8 @@ XDMFFile::~XDMFFile() { close(); }
 //-----------------------------------------------------------------------------
 void XDMFFile::close()
 {
+  flush();
```

Comment:
I don't think flush should be called here - no need to also flush the HDF5 file.